### PR TITLE
Use objects everywhere instead of static methods

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
Part 2 of 3. 

Relays on https://github.com/google/web-stories-wp/pull/2593 

Remove static methods, to make the code more testable.

## Relevant Technical Choices

Note the following 
Stories_Controller
 - Use discovery class. 

Story_Post_Tpye
- Use discovery class. 
- Use media class. 

Story_Renderer
- Use discovery class. 
- Use media class. 

In these cases, a create a new object using these cases. In the next PR, I will use dependency injection to remove this objects. 

## Testing Instructions

---

Fixes #

### CODE REVIEW

File: includes/Discovery.php [hunk 1]
```diff
@@ -306,14 +311,26 @@ public static function get_publisher_logo() {
...
+
+	/**
+	 * Publisher logo placeholder for static content output which will be replaced server-side.
+	 *
+	 * Uses a fallback logo to always create valid AMP in FE.
+	 *
+	 * @return string
+	 */
+	public function get_publisher_logo_placeholder() {
+		return WEBSTORIES_PLUGIN_DIR_URL . 'assets/images/fallback-wordpress-publisher-logo.png';
```
Comment (spacedmonkey) [inline includes/Discovery.php:323]: Should we have a filter here?

Comment (swissspidy) [reply includes/Discovery.php:323]: This is just the fallback if there is no publisher logo, as a last resort. If someone wants to override the publisher logo, they can use the `web_stories_publisher_logo` filter.

File: includes/Story_Post_Type.php [hunk 2]
```diff
@@ -254,7 +258,7 @@ public static function filter_use_block_editor_for_post_type( $use_block_editor,
 	 *
 	 * @return void
 	 */
-	public static function admin_enqueue_scripts( $hook ) {
+	public function admin_enqueue_scripts( $hook ) {
```
Comment (swissspidy) [inline includes/Story_Post_Type.php:261]: The `$settings = self::get_editor_settings();` call further down can be replaced with `$settings = self::get_editor_settings();`

File: includes/Media.php [hunk 3]
```diff
@@ -294,6 +273,60 @@ public static function rest_api_init() {
...
+	 * @return string
+	 */
+	public function get_callback_media_source( $prepared ) {
+		$id = $prepared['id'];
+
+		$terms = wp_get_object_terms( $id, self::STORY_MEDIA_TAXONOMY );
+		if ( is_array( $terms ) && $terms ) {
+			$term = array_shift( $terms );
+
+			return $term->slug;
```
Comment (swissspidy) [inline includes/Media.php:290]: Let's optimize this while we're at it:

```suggestion
			return array_shift( $terms )->slug;
```

File: includes/Media.php [hunk 4]
```diff
@@ -294,6 +273,60 @@ public static function rest_api_init() {
...
+	 *
+	 * @param array $prepared Prepared data before response.
+	 *
+	 * @return array
+	 */
+	public function get_callback_featured_media_src( $prepared ) {
+		$id    = $prepared['featured_media'];
+		$image = [];
+		if ( $id ) {
+			$image = self::get_thumbnail_data( $id );
```
Comment (swissspidy) [inline includes/Media.php]:
```suggestion
			return $this->get_thumbnail_data( $id );
```

File: includes/Media.php [hunk 5]
```diff
@@ -302,7 +335,7 @@ public static function rest_api_init() {
 	 *
 	 * @return array $response;
 	 */
-	public static function wp_prepare_attachment_for_js( $response, $attachment ) {
+	public function wp_prepare_attachment_for_js( $response, $attachment ) {
```
Comment (swissspidy) [inline includes/Media.php:338]: `$image = self::get_thumbnail_data( $thumbnail_id );` below can be replaced with `$image = $this->get_thumbnail_data( $thumbnail_id );`

File: includes/Media.php [hunk 6]
```diff
@@ -372,7 +405,7 @@ public static function delete_video_poster( $attachment_id ) {
 	 *
 	 * @return array List of allowed file types.
 	 */
-	public static function get_allowed_file_types() {
+	public function get_allowed_file_types() {
 		$allowed_mime_types = self::get_allowed_mime_types();
```
Comment (swissspidy) [inline includes/Media.php]:
```suggestion
		$allowed_mime_types = $this->get_allowed_mime_types();
```

File: includes/REST_API/Fonts_Controller.php [hunk 7]
```diff
@@ -79,7 +79,8 @@ public function register_routes() {
 	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
 	 */
 	public function get_items( $request ) {
-		$fonts       = Fonts::get_fonts();
+		$font        = new Fonts();
```
Comment (swissspidy) [inline includes/REST_API/Fonts_Controller.php]: `$font` is also used in the `foreach` loop below, which is confusing and error prone.

Suggestion:

Rename variables like so:

```php
$fonts = new Fonts();
$fonts_list = $font->get_fonts();
```

I think in the tests it's already like this, so this change would make things consistent.

**Review by swissspidy — COMMENTED** [review]
A few nits, but otherwise looks good 👍

File: includes/Media.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/Media.php:290]: I don't like that, it is not clear.

File: includes/REST_API/Fonts_Controller.php (hunk 7 above)
Comment (spacedmonkey) [reply includes/REST_API/Fonts_Controller.php]: Fixed.

File: includes/Media.php (hunk 4 above)
Comment (spacedmonkey) [reply includes/Media.php]: Fixed https://github.com/google/web-stories-wp/pull/2595/commits/88733befb0e22cb269bdf2cad75b7c249ce7aeb4

File: includes/Media.php (hunk 5 above)
Comment (spacedmonkey) [reply includes/Media.php:338]: Fixed https://github.com/google/web-stories-wp/pull/2595/commits/88733befb0e22cb269bdf2cad75b7c249ce7aeb4

File: includes/Media.php (hunk 6 above)
Comment (spacedmonkey) [reply includes/Media.php]: Fixed https://github.com/google/web-stories-wp/pull/2595/commits/88733befb0e22cb269bdf2cad75b7c249ce7aeb4

File: includes/Story_Post_Type.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Story_Post_Type.php:261]: Fixed https://github.com/google/web-stories-wp/pull/2595/commits/88733befb0e22cb269bdf2cad75b7c249ce7aeb4

File: includes/Media.php (hunk 3 above)
Comment (spacedmonkey) [reply includes/Media.php:290]: Really we should check the element has a slug before extracting it here.

Comment (swissspidy) [reply includes/Media.php:290]: Shouldn't this list only include `WP_Term` objects though?

**Review by swissspidy — APPROVED** [review]

**Pull request merged**